### PR TITLE
Refactor DPR for haystack compatibility

### PR DESCRIPTION
**Pull request by kolk**
- [x] TextSimilarityProcessor query/passage optional
- [x] BiAdaptiveModel query/passage forward pass optional
- [x] `max_seq_len_query` and `max_seq_len_context` added to TextSimilarityProcessor
- [x] Incorporate changes from PR: DPR transformers modelhub model load bugfix #595

### CODE REVIEW

File: farm/modeling/prediction_head.py [hunk 1]
```diff
@@ -1654,7 +1654,7 @@ def _embeddings_to_scores(self, query_vectors:torch.Tensor, context_vectors:torc
         softmax_scores = nn.functional.log_softmax(scores, dim=1)
         return softmax_scores
 
-    def logits_to_loss(self, logits: Tuple[torch.Tensor, torch.Tensor], **kwargs):
+    def logits_to_loss(self, logits: dict, **kwargs):
```
Comment (tholor) [inline farm/modeling/prediction_head.py:1657]: not sure if we should really switch to a dict here. It's kind of breaking the design with other prediction head / models in FARM

Comment (kolk) [reply farm/modeling/prediction_head.py]: reverted back to original

File: farm/data_handler/processor.py [hunk 2]
```diff
@@ -1867,10 +1870,12 @@ def __init__(
         self.num_positives = num_positives
         self.shuffle_negatives = shuffle_negatives
         self.shuffle_positives = shuffle_positives
+        self.max_seq_len_query = max_seq_len_query
+        self.max_seq_len_context = max_seq_len_context
 
         super(TextSimilarityProcessor, self).__init__(
             tokenizer=tokenizer,
-            max_seq_len=max_seq_len,
+            max_seq_len=max_seq_len_context,
```
[the remark this answers is not in the document]
Comment (kolk) [reply farm/data_handler/processor.py:1879]: fixed

File: farm/data_handler/processor.py [hunk 3]
```diff
@@ -1798,8 +1798,9 @@ def __init__(
         self,
         tokenizer,
         passage_tokenizer,
-        max_seq_len,
-        data_dir,
+        max_seq_len_query,
+        max_seq_len_context,
```
[the remark this answers is not in the document]
Comment (kolk) [reply farm/data_handler/processor.py]: done

**Review by tholor — APPROVED** [review]
Looking good. Once CI passes, we can merge this

**Pull request merged**